### DISCUSSION TITLE
Sqlite

### DISCUSSION THREAD

**Discussion by yogeshkumarmangal** [directus/directus#10856] [Archive]
How can data Fatch from sqlite directly in direct us

**Comment by azrikahar** [comment]
Directus provides both REST API and GraphQL endpoints for us to fetch data from our database, which in your case, SQLite. You can refer the docs for information: https://docs.directus.io/reference/introduction/

**Comment by yogeshkumarmangal** [comment]
Sir,My question is:-
1. I have data in csv  and Sqlite both in my system
2. How can direct data get in direct us from csv or sqlite in  a table
format.

**Reply by azrikahar** [reply]
If I understand this correctly, you should be able to connect Directus to your SQLite database and it should be able to read your existing data there. 👍

As for importing CSV data (once again this is assuming I interpreted your question correctly), you can do so with the import utility endpoint: https://docs.directus.io/reference/system/utilities/#import-data-from-file

**Comment by yogeshkumarmangal** [comment]
Awesome, thanks!

**Comment by yogeshkumarmangal** [comment]
Can we get proper video this will very helpful for me and others

**Comment by yogeshkumarmangal** [comment]
how can install direct us initial table in empty database (sqlite) using
bootstrap?

**Reply by azrikahar** [reply]
You can run `npx directus bootstrap` ([reference to docs for this command](https://docs.directus.io/reference/cli/#bootstrap-a-project)) in your Directus project. Do make sure `DB_CLIENT` is set as `sqlite3` if it's not, and set `DB_FILENAME` to your sqlite3 file. ([reference for these variables](https://docs.directus.io/configuration/config-options/#database))

An example filename on Windows would be: `DB_FILENAME="C:\Users\Azri\Documents\Projects\my-directus\data.db"`, with `my-directus` being my Directus project.

**Reply by Sleepful** [reply]
https://docs.directus.io/self-hosted/quickstart.html#create-a-docker-compose-file on point

**Comment by yogeshkumarmangal** [comment]
sorry ,but i did not understand
How to create table in my database  db.sqlite
can you more explore this ?

**Reply by Pachat** [reply]
Just create a Collection in Directus / Settings / Data Model : the name of the Collection will be the name of the table that will be  instantly created on saving

**Comment by yogeshkumarmangal** [comment]
Got it, thanks!
How can make a graph in dashboard in direct us

**Comment by yogeshkumarmangal** [comment]
Sir in Direct us in dashboard not showing any graph

**Reply by azrikahar** [reply]
Would you mind elaborating on the situation here? Does it mean you have configured the panels in your dashboard and they are not showing up, or perhaps you were expecting some default dashboards out of the box?

You can also refer the Insights module docs in case you have any doubts on setting it up: https://docs.directus.io/app/insights/

**Comment by yogeshkumarmangal** [comment]
Sir , Without Primary Key we can not see data in Direct us Server.

**Reply by Pachat** [reply]
No. A primary key for each collection is a requirement [in Directus](https://docs.directus.io/getting-started/glossary/#primary-key-pk).

**Comment by yogeshkumarmangal** [comment]
Ok, thanks.
Time series plot in direct us Dashboard
what is date format in this ?
